### PR TITLE
feature (ref 33679): create a column for original statement in xlx fi…

### PR DESCRIPTION
**Pull request by jalalifardemos**
**Ticket:** https://yaits.demos-deutschland.de/T33579

Description: add original statement as a xlsx column in zip export

### Linked PRs (optional)
List other PRs that are somehow connected to this and explain the connection.

- https://github.com/demos-europe/demosplan-project-regio/pull/91

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php [hunk 1]
```diff
@@ -101,25 +106,45 @@ private function fillZipWithData(ZipStream $zipStream, array $file): void
...
-                } catch (InvalidDataException $e) {
-                    $this->handleError($e, self::FILE_HASH_INVALID);
-                    ++$this->errorCount['attachmentNotAddedCount'];
-                } catch (Exception $e) {
-                    $this->handleError($e, self::UNKOWN_ERROR);
-                    ++$this->errorCount['attachmentUnkownErrorCount'];
                 }
+                /**
+                 * The originalAttachment can be of type { @see File } if there was a stn attachment already set within
+                 * the exportet entity. Or it is an array in case the { @see AssessmentTablePdfExporter } got invoked
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php:120]: typo exportet

File: demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTablePdfExporter.php [hunk 2]
```diff
@@ -113,7 +113,17 @@ public function __invoke(array $parameters): array
...
-            $parameters = $this->addStatementsFromCurrentQueryHashToFilter($parameters, $procedureId, $original);
+            if (!array_key_exists('statementId', $parameters)) {
+                $parameters = $this->addStatementsFromCurrentQueryHashToFilter($parameters, $procedureId, $original);
+            } else {
+                /*
+                 * in case the key 'statementId' was set by the invoking
+                 * { @link AssessmentTableZipExporter::getAttachmentsOfStatements }
+                 * do not try to obtain the ids from session
+                 */
+                $parameters['items'] = [$parameters['statementId']];
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTablePdfExporter.php:124]: Param `sort` is not set here. Is this intended?

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php [hunk 3]
```diff
@@ -101,25 +106,45 @@ private function fillZipWithData(ZipStream $zipStream, array $file): void
...
+                if ($originalAttachment instanceof File) {
+                    $this->zipExportService->addFileToZipStream(
+                        $originalAttachment->getFilePathWithHash(),
+                        $file['zipFileName'].'/'.$originalAttachment->getHash().'_'.$originalAttachment->getFilename(),
+                        $zipStream
+                    );
+                }
+                if (is_array($originalAttachment)) {
+                    $zipStream->addFile(
+                        Utf8::toAscii(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php:133]: Library Utf8 is deprecated. Use symfony String component here

File: demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTableZipExporter.php [hunk 4]
```diff
@@ -110,18 +114,55 @@ private function getAttachmentsOfStatements(array $statementIds): array
...
+                $files[$index]['attachments'][] = $statementAttachment->getFile();
+            }
+            // set the stn attachment:
+            // if present just take the given one.
+            $files[$index]['originalAttachment'] = $this->getOriginalAttachment($statementId);
+            if (null === $files[$index]['originalAttachment']) {
+                // if not present yet, invoke the pdfCreator and create an original-stn-pdf to use instead
+                $parameters['statementId'] =
+                    $this->statementService->getStatement($statementId)?->getOriginal()->getId();
+                $files[$index]['originalAttachment'] = $this->pdfExporter->__invoke(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTableZipExporter.php:140]: `__invoke` is not meant to be called directly

File: translations/messages+intl-icu.de.yml [hunk 5]
```diff
@@ -3110,6 +3110,7 @@ statement.assignment.override: "Stellungnahme mir zuweisen"
 statement.assignment.release: "Stellungnahme freigeben"
 statement.assignment.unassigned: "Aktuell nicht in Bearbeitung"
 statement.attachments.reference: "Referenzen auf Anhänge"
+statement.original.attachment.reference: "Referenzen auf original Anhang"
```
Comment (mussbach) [inline translations/messages+intl-icu.de.yml:3113]: Not so clear what this should be. When this should be kept it should be at least `Originalanhang`

**Review by mussbach — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTableZipExporter.php (hunk 4 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTableZipExporter.php]: done in [0e6e905](https://github.com/demos-europe/demosplan-core/pull/2444/commits/0e6e9057ec8a3988a461cea7b520254a1f7fe5ef)

File: translations/messages+intl-icu.de.yml (hunk 5 above)
Comment (jalalifardemos) [reply translations/messages+intl-icu.de.yml]: done in [ba0023b](https://github.com/demos-europe/demosplan-core/pull/2444/commits/ba0023badff5ef07ebf1b134aabd7d377c15dc6e)

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php (hunk 3 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php]: done in [0da0131](https://github.com/demos-europe/demosplan-core/pull/2444/commits/0da0131d53f7a23a10bc0e06ed2406cb32058a0f)

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php (hunk 1 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php]: done in [5b12050](https://github.com/demos-europe/demosplan-core/pull/2444/commits/5b120501f9de24a0a2ad3fcc1193a6483f9851bd)

File: translations/messages+intl-icu.de.yml (hunk 5 above)
Comment (MoritzMandler) [reply translations/messages+intl-icu.de.yml]: I need to know the final outcome to adjust this in https://github.com/demos-europe/demosplan-core/pull/2363
pls notify :)

File: demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTablePdfExporter.php (hunk 2 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTablePdfExporter.php:128]: done in [c576805](https://github.com/demos-europe/demosplan-core/pull/2444/commits/c5768052f92bda399a1945ead9598157b4cc75e2)

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php (hunk 1 above)
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php]: `typo` slipped in as well :-) Should be `within the exported entity`

File: translations/messages+intl-icu.de.yml (hunk 5 above)
Comment (mussbach) [reply translations/messages+intl-icu.de.yml]: I woud then suggest only `Originalanhang`

Comment (jalalifardemos) [reply translations/messages+intl-icu.de.yml]: done in [45f9819](https://github.com/demos-europe/demosplan-core/pull/2444/commits/45f98190a2614e931ec22813a601e7051689e075)

File: demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php (hunk 1 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php]: done in [45f9819](https://github.com/demos-europe/demosplan-core/pull/2444/commits/45f98190a2614e931ec22813a601e7051689e075)

File: translations/messages+intl-icu.de.yml (hunk 5 above)
Comment (mussbach) [reply translations/messages+intl-icu.de.yml]: @MoritzMandler `Originalanhang` it is then

**Review by mussbach — APPROVED** [review]
lgtm, codereview only.

**Pull request merged**